scripts/scene_elements/controllers/post_combat_controller.py

```diff
@@ -1,67 +1,60 @@
+from scripts.core.debug import Timer
 from scripts.scene_elements.troupe import Troupe
 from scripts.scene_elements.unit import Unit
-from scripts.scenes.post_combat.ui import PostCombatUI
 
 if TYPE_CHECKING:
-    from typing import Optional
-
     from scripts.core.game import Game
+    from scripts.scenes.world.scene import WorldScene
 
-__all__ = ["PostCombatScene"]
+__all__ = ["PostCombatController"]
 
 
-class PostCombatScene(Scene):
-    """
-    Handles RewardScene interactions and consolidates the rendering. RewardScene is used to provide a choice of
-    rewards for the player to pick from.
+class PostCombatController(Controller):
     """
+    Handles RewardScene interactions
 
-    def __init__(self, game: Game):
-        # start timer
-        start_time = time.time()
+    RewardScene is used to provide a choice of rewards for the player
+    to pick from.
 
-        super().__init__(game, SceneType.POST_COMBAT)
+    * Modify game state in accordance with game rules
+    * Do not draw anything
 
-        self.ui: PostCombatUI = PostCombatUI(game, self)
+    """
 
-        self.state: PostCombatState = PostCombatState.VICTORY
+    def __init__(self, game: Game, parent_scene: WorldScene):
+        with Timer("PostCombatController initialised"):
+            super().__init__(game, parent_scene)
+            self.state: PostCombatState = PostCombatState.VICTORY
 
-        # reward management
-        self.current_rewards = None  # holds the current rewards, e.g. troupe_rewards
-        self.reward_type: RewardType = RewardType.UNIT
-        self.num_rewards: int = 3
+            # reward management
+            # holds the current rewards, e.g. troupe_rewards
+            self.current_rewards = None
+            self.reward_type: RewardType = RewardType.UNIT
+            self.num_rewards: int = 3
 
-        # reward options
-        self.gold_reward: int = 0
-
-        self.troupe_rewards: Optional[Troupe] = None
-        self.resource_rewards = None
-        self.upgrade_rewards = None
-        self.action_rewards = None
-
-        # record duration
-        end_time = time.time()
-        logging.debug(f"RewardScene: initialised in {format(end_time - start_time, '.2f')}s.")
+            # reward options
+            self.gold_reward: int = 0
+            self.troupe_rewards: Optional[Troupe] = None
+            self.resource_rewards = None
+            self.upgrade_rewards = None
+            self.action_rewards = None
 
     def update(self, delta_time: float):
-        super().update(delta_time)
-        self.ui.update(delta_time)
+        pass
 
     def reset(self):
-        self.ui = PostCombatUI(self._game, self)
-
         self.state = PostCombatState.VICTORY
 
         # reward management
-        self.current_rewards = None  # holds the current rewards, e.g. troupe_rewards
+        # holds the current rewards, e.g. troupe_rewards
+        self.current_rewards = None
```

Comment:
The whole reward system likely needs reviewing. Suggest for now we keep it simple and add handling for gold and a new unit.

Reply:
sure.  should that be part of this pr, or can we open a new issue to track that?